parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetWriter.java

```diff
@@ -50,6 +46,31 @@ protected ParquetValueWriter<T> createWriter(MessageType type) {
   protected abstract ParquetValueWriters.StructWriter<T> createStructWriter(
       List<ParquetValueWriter<?>> writers);
 
+  protected ParquetValueWriter<?> fixedWriter(ColumnDescriptor desc) {
+    return new GenericParquetWriter.FixedWriter(desc);
+  }
+
+  protected ParquetValueWriter<?> dateWriter(ColumnDescriptor desc) {
+    return new GenericParquetWriter.DateWriter(desc);
+  }
+
+  protected ParquetValueWriter<?> timeWriter(ColumnDescriptor desc) {
+    return new GenericParquetWriter.TimeWriter(desc);
+  }
+
+  protected ParquetValueWriter<?> timestampWriter(ColumnDescriptor desc, boolean isAdjustedToUTC) {
+    if (isAdjustedToUTC) {
+      return new GenericParquetWriter.TimestamptzWriter(desc);
+    } else {
+      return new GenericParquetWriter.TimestampWriter(desc);
+    }
+  }
+
+  protected ParquetValueWriter<?> uuidWriter(ColumnDescriptor desc) {
+    // Use primitive-type writer (as FIXED_LEN_BYTE_ARRAY); no special writer needed.
+    return null;
```

Comment:
I think I commented on this in the last round of reviews. This isn't correct. Incoming values are of type `UUID` so this needs a writer that can convert UUID into a byte array. This should return `ParquetValueWriter.uuids(desc)`.

There's also no need to add a method for this because it is the same between the generic and internal object models.

Reply:
Sorry, the existing testcases was the reason for confusion as I mentioned in https://github.com/apache/iceberg/pull/11904#discussion_r1926274339

I will update the testcases of Arrow too in this PR.